![example](images/director_shot.jpeg)

# Microsoft Movie Analysis

**Authors:** Albane Colmenares
***

## Overview

A one-paragraph overview of the project, including the business problem, data, methods, results and recommendations.

## Business Problem

Summary of the business problem you are trying to solve, and the data questions that you plan to answer to solve them.

***
Questions to consider:
* What are the business's pain points related to this project?
* How did you pick the data analysis question(s) that you did?
* Why are these questions important from a business perspective?
***

In [1]:
# Understand:  
# films's types that are doing best at box office. 
# Goal: 
# Become a new movie studio 
# does studio type matter?

## Data Understanding

Describe the data being used for this project.
***
Questions to consider:
* Where did the data come from, and how do they relate to the data analysis questions?
* What do the data represent? Who is in the sample and what variables are included?
* What is the target variable?
* What are the properties of the variables you intend to use?
***

In [2]:
# Import standard packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import sqlite3 

%matplotlib inline


In [3]:
# Here you run your code to explore the data


### 1. bom.movie_gross.csv.gz

In [4]:
# Loading and inspecting available datasets
# Loading bom.movie_gross and storing data into df_movie_gross
df_movie_gross = pd.read_csv('data/bom.movie_gross.csv.gz', compression='gzip')

df_movie_gross.head()


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [5]:
# conn = sqlite3.connect('data/im.db')

# df_imdbtest = pd.read_sql('data/im.db.zip', conn)
# #
# df_imdbtest.head()

In [6]:
# Inspect overall shape and info of the dataframe
# df_movie_gross.shape
# df_movie_gross.info()

In [7]:
# df_movie_gross data cleaning
# Inspect the 5 studio null information
# df_movie_gross[df_movie_gross['studio'].isnull()]


In [8]:
# Inspect the studios in general
# studios = df_movie_gross.drop_duplicates(subset=['studio'])
# print(studios['studio'].tolist())
# print(len(studios['studio'].tolist()))
# There are 258 production studios, which is large enough to remove 5 
# rows that don't have any studios. 

In [9]:
# Dropping rows
# df_movie_gross = df_movie_gross[df_movie_gross['studio'].notna()]
# Verifying that na rows were dropped 
# df_movie_gross.info()

In [10]:
# Inspecting movies that don't have domestic revenue: do they have a 
# foreign revenue? 
# df_movie_gross[df_movie_gross['domestic_gross'].isnull()]

In [11]:
# All movies that don't have domestic revenue were distributed oversees 

In [12]:
# Now inspecting the same info for foreign revenue
# df_movie_gross[df_movie_gross['foreign_gross'].isnull()]

In [13]:
# Convert foreign_gross column as float
# df_movie_gross['foreign_gross'] = df_movie_gross['foreign_gross'].str.replace(',','').astype(np.float64)

# Filling na values with 0 on both columns:
# df_movie_gross.update(df_movie_gross[['domestic_gross', 'foreign_gross']].fillna(0))

# df_movie_gross.info()

In [14]:
# # Create a column total gross 
# df_movie_gross["total_gross"] = df_movie_gross["domestic_gross"] + df_movie_gross["foreign_gross"]
# df_movie_gross["total_gross (in '000)"] = (df_movie_gross["domestic_gross"] + df_movie_gross["foreign_gross"]) / 1000

In [15]:
# # Sorting by highest total revenue generated by movies
# df_movie_gross.sort_values(by=["total_gross"], ascending=False, inplace=True)
# df_movie_gross

In [16]:
# top_20_revenue_movies = df_movie_gross.head(20).reset_index(drop=True)
# top_20_revenue_movies

In [17]:
# bottom_20_revenue_movies = df_movie_gross.tail(20).reset_index(drop=True)
# bottom_20_revenue_movies

In [18]:
# # Reviewing how frequently movies generate large revenues

# df_movie_gross["total_gross"].hist(bins=10)
# plt.xlabel("Total Revenue in '000")
# plt.ylabel("Number of movies")
# plt.title("Distribution of films' revenues")

In [19]:
# The histogram is highly skewed to the left, indicating the largest majority of movies make between $100 and $1,300. 
# This emphasizes that movies making high revenues are all the more unique

## Data Exploration


### 2. im.db.zip

In [20]:
# Loading and inspecting available datasets
# Loading bom.movie_gross and storing data into df_movie_gross

import zipfile
with zipfile.ZipFile('data/im.db.zip', 'r') as zip_ref:
    zip_ref.extractall('data')


In [21]:
# Loading im.db and storing data into df_imdb

# Creating connection to database
conn = sqlite3.connect('data/im.db')

# Creating a cursor
cur = conn.cursor()


In [23]:
imdb = pd.read_sql("""
                SELECT * 
                FROM sqlite_master
                ;
""", con=conn)
imdb

type           name       tbl_name  rootpage  \
0  table   movie_basics   movie_basics         2   
1  table      directors      directors         3   
2  table      known_for      known_for         4   
3  table     movie_akas     movie_akas         5   
4  table  movie_ratings  movie_ratings         6   
5  table        persons        persons         7   
6  table     principals     principals         8   
7  table        writers        writers         9   

                                                 sql  
0  CREATE TABLE "movie_basics" (\n"movie_id" TEXT...  
1  CREATE TABLE "directors" (\n"movie_id" TEXT,\n...  
2  CREATE TABLE "known_for" (\n"person_id" TEXT,\...  
3  CREATE TABLE "movie_akas" (\n"movie_id" TEXT,\...  
4  CREATE TABLE "movie_ratings" (\n"movie_id" TEX...  
5  CREATE TABLE "persons" (\n"person_id" TEXT,\n ...  
6  CREATE TABLE "principals" (\n"movie_id" TEXT,\...  
7  CREATE TABLE "writers" (\n"movie_id" TEXT,\n  ...

In [24]:
# movie_basics exploration

movie_basics = pd.read_sql(
"""
SELECT * 
FROM movie_basics
;
"""
, con=conn)

movie_basics.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [76]:
movie_basics.info()

# start_year will be used to merge the dataframes and is an integer type: 
# will convert the string to integrer type in df_movie_budgets



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


In [25]:
# directors exploration

directors_df = pd.read_sql(
"""
SELECT * 
FROM directors
;
"""
, con=conn)

directors_df

movie_id   person_id
0       tt0285252   nm0899854
1       tt0462036   nm1940585
2       tt0835418   nm0151540
3       tt0835418   nm0151540
4       tt0878654   nm0089502
...           ...         ...
291169  tt8999974  nm10122357
291170  tt9001390   nm6711477
291171  tt9001494  nm10123242
291172  tt9001494  nm10123248
291173  tt9004986   nm4993825

[291174 rows x 2 columns]

In [26]:
# known_for exploration

known_for_df = pd.read_sql(
"""
SELECT * 
FROM known_for
;
"""
, con=conn)

known_for_df

person_id   movie_id
0        nm0061671  tt0837562
1        nm0061671  tt2398241
2        nm0061671  tt0844471
3        nm0061671  tt0118553
4        nm0061865  tt0896534
...            ...        ...
1638255  nm9990690  tt9090932
1638256  nm9990690  tt8737130
1638257  nm9991320  tt8734436
1638258  nm9991320  tt9615610
1638259  nm9993380  tt8743182

[1638260 rows x 2 columns]

In [27]:
# movie_akas exploration

movie_akas_df = pd.read_sql(
"""
SELECT * 
FROM movie_akas
;
"""
, con=conn)

movie_akas_df

movie_id  ordering                                    title region  \
0       tt0369610        10                            Джурасик свят     BG   
1       tt0369610        11                        Jurashikku warudo     JP   
2       tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3       tt0369610        13                  O Mundo dos Dinossauros     BR   
4       tt0369610        14                           Jurassic World     FR   
...           ...       ...                                      ...    ...   
331698  tt9827784         2                       Sayonara kuchibiru   None   
331699  tt9827784         3                            Farewell Song    XWW   
331700  tt9880178         1                              La atención   None   
331701  tt9880178         2                              La atención     ES   
331702  tt9880178         3                            The Attention    XWW   

       language        types   attributes  is_original_title  
0            bg         None         None                0.0  
1          None  imdbDisplay         None                0.0  
2          None  imdbDisplay         None                0.0  
3          None         None  short title                0.0  
4          None  imdbDisplay         None                0.0  
...         ...          ...          ...                ...  
331698     None     original         None                1.0  
331699       en  imdbDisplay         None                0.0  
331700     None     original         None                1.0  
331701     None         None         None                0.0  
331702       en  imdbDisplay         None                0.0  

[331703 rows x 8 columns]

In [28]:
# movie_id for Marvel's The Avengers where original_title = 1 
movie_akas_df[(movie_akas_df["title"].str.contains("Avengers")) & (movie_akas_df["is_original_title"] == 1)]

movie_id  ordering                    title region language  \
69881   tt2395427        24  Avengers: Age of Ultron   None     None   
81106   tt0848228        19             The Avengers   None     None   
190575  tt4154796        15        Avengers: Endgame   None     None   
311338  tt4154756        30   Avengers: Infinity War   None     None   

           types attributes  is_original_title  
69881   original       None                1.0  
81106   original       None                1.0  
190575  original       None                1.0  
311338  original       None                1.0

In [29]:
# what are region and attributes

# region
movie_akas_df[movie_akas_df["region"].notnull()]


movie_id  ordering                                    title region  \
0       tt0369610        10                            Джурасик свят     BG   
1       tt0369610        11                        Jurashikku warudo     JP   
2       tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3       tt0369610        13                  O Mundo dos Dinossauros     BR   
4       tt0369610        14                           Jurassic World     FR   
...           ...       ...                                      ...    ...   
331695  tt9755806         2                            Большая Акула     RU   
331697  tt9827784         1                       Sayonara kuchibiru     JP   
331699  tt9827784         3                            Farewell Song    XWW   
331701  tt9880178         2                              La atención     ES   
331702  tt9880178         3                            The Attention    XWW   

       language        types   attributes  is_original_title  
0            bg         None         None                0.0  
1          None  imdbDisplay         None                0.0  
2          None  imdbDisplay         None                0.0  
3          None         None  short title                0.0  
4          None  imdbDisplay         None                0.0  
...         ...          ...          ...                ...  
331695     None         None         None                0.0  
331697     None         None         None                0.0  
331699       en  imdbDisplay         None                0.0  
331701     None         None         None                0.0  
331702       en  imdbDisplay         None                0.0  

[278410 rows x 8 columns]

In [30]:
# attributes

movie_akas_df[movie_akas_df["attributes"].notnull()]

movie_id  ordering                    title region language types  \
3       tt0369610        13  O Mundo dos Dinossauros     BR     None  None   
12      tt0369610        21        Jurassic World 3D     US     None  None   
13      tt0369610        22        Jurassic World 3D     DE     None  None   
14      tt0369610        23        Jurassic World 3D    XWW     None  None   
21      tt0369610         2                 Ebb Tide     US     None  None   
...           ...       ...                      ...    ...      ...   ...   
331578  tt8899108         2         Volviendo a casa     UY     None  None   
331590  tt8997034         1         Little Germanics    XWW       en  None   
331633  tt9190740         1           Car Ribna Wali     IN       ur  None   
331660  tt9447768         1          The Town School     LK     None  None   
331666  tt9486128         3       We'll Rise at Down    XWW       en  None   

                            attributes  is_original_title  
3                          short title                0.0  
12                         3-D version                0.0  
13                         3-D version                0.0  
14                         3-D version                0.0  
21                  fake working title                0.0  
...                                ...                ...  
331578      original subtitled version                0.0  
331590  informal literal English title                0.0  
331633     alternative transliteration                0.0  
331660     alternative transliteration                0.0  
331666           literal English title                0.0  

[14925 rows x 8 columns]

In [31]:
# movie_ratings exploration

movie_ratings_df = pd.read_sql(
"""
SELECT * 
FROM movie_ratings
;
"""
, con=conn)

movie_ratings_df

movie_id  averagerating  numvotes
0      tt10356526            8.3        31
1      tt10384606            8.9       559
2       tt1042974            6.4        20
3       tt1043726            4.2     50352
4       tt1060240            6.5        21
...           ...            ...       ...
73851   tt9805820            8.1        25
73852   tt9844256            7.5        24
73853   tt9851050            4.7        14
73854   tt9886934            7.0         5
73855   tt9894098            6.3       128

[73856 rows x 3 columns]

In [32]:
# persons exploration

persons_df = pd.read_sql(
"""
SELECT * 
FROM persons
;
"""
, con=conn)

persons_df


person_id         primary_name  birth_year  death_year  \
0       nm0061671    Mary Ellen Bauder         NaN         NaN   
1       nm0061865         Joseph Bauer         NaN         NaN   
2       nm0062070           Bruce Baum         NaN         NaN   
3       nm0062195         Axel Baumann         NaN         NaN   
4       nm0062798          Pete Baxter         NaN         NaN   
...           ...                  ...         ...         ...   
606643  nm9990381         Susan Grobes         NaN         NaN   
606644  nm9990690          Joo Yeon So         NaN         NaN   
606645  nm9991320       Madeline Smith         NaN         NaN   
606646  nm9991786  Michelle Modigliani         NaN         NaN   
606647  nm9993380       Pegasus Envoyé         NaN         NaN   

                                      primary_profession  
0              miscellaneous,production_manager,producer  
1             composer,music_department,sound_department  
2                             miscellaneous,actor,writer  
3       camera_department,cinematographer,art_department  
4       production_designer,art_department,set_decorator  
...                                                  ...  
606643                                           actress  
606644                                           actress  
606645                                           actress  
606646                                          producer  
606647                             director,actor,writer  

[606648 rows x 5 columns]

In [33]:
persons_df["primary_name"].unique()

array(['Mary Ellen Bauder', 'Joseph Bauer', 'Bruce Baum', ...,
       'Joo Yeon So', 'Michelle Modigliani', 'Pegasus Envoyé'],
      dtype=object)

In [34]:
# principals exploration
principals_df = pd.read_sql(
"""
SELECT * 
FROM principals
;
"""
, con=conn)

principals_df

movie_id  ordering   person_id  category       job  \
0        tt0111414         1   nm0246005     actor      None   
1        tt0111414         2   nm0398271  director      None   
2        tt0111414         3   nm3739909  producer  producer   
3        tt0323808        10   nm0059247    editor      None   
4        tt0323808         1   nm3579312   actress      None   
...            ...       ...         ...       ...       ...   
1028181  tt9692684         1   nm0186469     actor      None   
1028182  tt9692684         2   nm4929530      self      None   
1028183  tt9692684         3  nm10441594  director      None   
1028184  tt9692684         4   nm6009913    writer    writer   
1028185  tt9692684         5  nm10441595  producer  producer   

                   characters  
0                 ["The Man"]  
1                        None  
2                        None  
3                        None  
4            ["Beth Boothby"]  
...                       ...  
1028181  ["Ebenezer Scrooge"]  
1028182   ["Herself","Regan"]  
1028183                  None  
1028184                  None  
1028185                  None  

[1028186 rows x 6 columns]

In [35]:
# writers exploration

writers_df = pd.read_sql(
"""
SELECT * 
FROM writers
;
"""
, con=conn)

writers_df

movie_id   person_id
0       tt0285252   nm0899854
1       tt0438973   nm0175726
2       tt0438973   nm1802864
3       tt0462036   nm1940585
4       tt0835418   nm0310087
...           ...         ...
255868  tt8999892  nm10122246
255869  tt8999974  nm10122357
255870  tt9001390   nm6711477
255871  tt9004986   nm4993825
255872  tt9010172   nm8352242

[255873 rows x 2 columns]

### 3. rt.movie_info.tsv

In [36]:
# Inspecting rt.movie_info file
# Loading rt.movie_info and storing data into df_movie_info

df_movie_info = pd.read_csv('data/rt.movie_info.tsv.gz', compression='gzip', sep='\t')

df_movie_info.head()


id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [37]:
# are there other currencies than box office? 

df_movie_info[df_movie_info["currency"].notnull()]["currency"].unique()

# no - only $

array(['$'], dtype=object)

In [38]:
# Inspect overall shape and info of the dataframe
df_movie_info.shape
df_movie_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


In [39]:
# missing revenues in box office in df_movie_info but is it the same info in the first table

In [40]:
df_movie_info[df_movie_info["box_office"].notnull()]

id                                           synopsis rating  \
1        3  New York City, not-too-distant-future: Eric Pa...      R   
6       10  Some cast and crew from NBC's highly acclaimed...  PG-13   
7       13  Stewart Kane, an Irishman living in the Austra...      R   
8       14  "Love Ranch" is a bittersweet love story that ...      R   
15      22  Two-time Academy Award Winner Kevin Spacey giv...      R   
...    ...                                                ...    ...   
1541  1980  A band of renegades on the run in outer space ...  PG-13   
1542  1981  Money, Fame and the Knowledge of English. In I...     NR   
1545  1985  A woman who joins the undead against her will ...      R   
1546  1986  Aki Kaurismaki's The Man Without a Past opens ...     PG   
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   

                                                 genre             director  \
1                    Drama|Science Fiction and Fantasy     David Cronenberg   
6                                               Comedy          Jake Kasdan   
7                                                Drama         Ray Lawrence   
8                                                Drama      Taylor Hackford   
15                   Comedy|Drama|Mystery and Suspense  George Hickenlooper   
...                                                ...                  ...   
1541  Action and Adventure|Science Fiction and Fantasy          Joss Whedon   
1542                                      Comedy|Drama         Gauri Shinde   
1545                       Horror|Mystery and Suspense  Sebastian Gutierrez   
1546          Art House and International|Comedy|Drama                  NaN   
1555  Action and Adventure|Horror|Mystery and Suspense                  NaN   

                                writer  theater_date      dvd_date currency  \
1         David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
6                           Mike White  Jan 11, 2002  Jun 18, 2002        $   
7     Raymond Carver|Beatrix Christian  Apr 27, 2006   Oct 2, 2007        $   
8                        Mark Jacobson  Jun 30, 2010   Nov 9, 2010        $   
15                       Norman Snider  Dec 17, 2010   Apr 5, 2011        $   
...                                ...           ...           ...      ...   
1541                       Joss Whedon  Sep 30, 2005  Dec 20, 2005        $   
1542                      Gauri Shinde   Oct 5, 2012  Nov 20, 2012        $   
1545               Sebastian Gutierrez   Jun 1, 2007   Oct 9, 2007        $   
1546                               NaN  Aug 30, 2002   Oct 7, 2003        $   
1555                               NaN  Aug 18, 2006   Jan 2, 2007        $   

      box_office      runtime                  studio  
1        600,000  108 minutes       Entertainment One  
6     41,032,915   82 minutes      Paramount Pictures  
7        224,114  123 minutes  Sony Pictures Classics  
8        134,904  117 minutes                     NaN  
15     1,039,869  108 minutes            ATO Pictures  
...          ...          ...                     ...  
1541  25,335,935  119 minutes      Universal Pictures  
1542   1,416,189  129 minutes      Eros Entertainment  
1545      59,371   98 minutes        IDP Distribution  
1546     794,306   97 minutes                     NaN  
1555  33,886,034  106 minutes         New Line Cinema  

[340 rows x 12 columns]

## notes
### df_movie_info data cleaning: 
-  all dates are strings, turn into datetime, keep only year? create month and year columns? 
-  only 340 rows with currency and box office, can we do anything with this information?
-  runtime should be useful to test correlation between rating or reviews, so turn into integer 

### 4. rt.reviews.tsv

In [41]:
# Inspecting rt.reviews file
# Loading rt.reviews and storing data into df_reviews

df_reviews = pd.read_csv('data/rt.reviews.tsv.gz', compression='gzip', sep='\t', encoding = 'unicode_escape')
df_reviews.head()


id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [42]:
# Inspect overall shape and info of the dataframe
df_reviews.shape
df_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


## notes
-  review null rows and either drop or replace
-  is rating out of 5 for all and can one same ratio be used for all

### 5. tmdb.movies.csv

In [43]:
# Inspecting tmdb.movies file
# Loading tmdb.movies and storing data into df_tmdb_movies
# Dropping the unnamed column as well 


df_tmdb_movies = pd.read_csv('data/tmdb.movies.csv.gz', compression='gzip', index_col=0)
df_tmdb_movies.head()

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [44]:
# df_tmdb_movies.genre_ids[0]
df_tmdb_movies.genre_ids = df_tmdb_movies.genre_ids.apply(lambda x: x[1:-1].split(','))
df_tmdb_movies.head()

genre_ids     id original_language  \
0       [12,  14,  10751]  12444                en   
1  [14,  12,  16,  10751]  10191                en   
2         [12,  28,  878]  10138                en   
3       [16,  35,  10751]    862                en   
4         [28,  878,  12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [135]:
df_tmdb_movies[df_tmdb_movies["original_title"].str.contains("(?i)avatar")]

genre_ids      id original_language  \
6      [28,  12,  14,  878]   19995                en   
1831                     []  278698                en   
3387        [878,  27,  53]   79582                en   
23157          [12,  10751]  460441                en   

                 original_title  popularity release_date  \
6                        Avatar      26.526   2009-12-18   
1831             Avatar Spirits       0.600   2010-06-22   
3387         Aliens vs. Avatars       2.199   2011-09-20   
23157  Avatar Flight of Passage       0.600   2017-05-01   

                          title  vote_average  vote_count  
6                        Avatar           7.4       18676  
1831             Avatar Spirits           9.5           2  
3387         Aliens vs. Avatars           2.4          12  
23157  Avatar Flight of Passage          10.0           1

In [45]:
type(df_tmdb_movies.genre_ids[0])

list

In [46]:
df_tmdb_movies.explode('genre_ids')


genre_ids      id original_language  \
0            12   12444                en   
0            14   12444                en   
0         10751   12444                en   
1            14   10191                en   
1            12   10191                en   
...         ...     ...               ...   
26515     10751  366854                en   
26515        12  366854                en   
26515        28  366854                en   
26516        53  309885                en   
26516        27  309885                en   

                                     original_title  popularity release_date  \
0      Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
0      Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
0      Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                          How to Train Your Dragon      28.734   2010-03-26   
1                          How to Train Your Dragon      28.734   2010-03-26   
...                                             ...         ...          ...   
26515                                  Trailer Made       0.600   2018-06-22   
26515                                  Trailer Made       0.600   2018-06-22   
26515                                  Trailer Made       0.600   2018-06-22   
26516                                    The Church       0.600   2018-10-05   
26516                                    The Church       0.600   2018-10-05   

                                              title  vote_average  vote_count  
0      Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
0      Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
0      Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                          How to Train Your Dragon           7.7        7610  
1                          How to Train Your Dragon           7.7        7610  
...                                             ...           ...         ...  
26515                                  Trailer Made           0.0           1  
26515                                  Trailer Made           0.0           1  
26515                                  Trailer Made           0.0           1  
26516                                    The Church           0.0           1  
26516                                    The Church           0.0           1  

[47834 rows x 9 columns]

If needed to change genre ids, use: explode. 


In [47]:
# Inspect overall shape and info of the dataframe
df_tmdb_movies.shape
df_tmdb_movies.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 3.3+ MB


## notes
-  unnamed0 column to drop --> example heroes_df = heroes_df.drop(['Unnamed: 0'], axis=1)
-  genre_ids to remove from list? 
-  release date: turn into datetime to create 2 columns for year and month?

### 6. tn.movie_budgets.csv

In [50]:
# Inspecting tn.movie_budgets file
# Loading tn.movie_budgets and storing data into df_movie_budgets

df_movie_budgets = pd.read_csv('data/tn.movie_budgets.csv.gz', compression='gzip')

df_movie_budgets.head()


id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [51]:
df_movie_budgets.shape
df_movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [52]:
# Make all number columns as integers and create year column


In [53]:
# Make all number columns as integers
columns_to_integers = ["production_budget" , "domestic_gross", "worldwide_gross"]

for column in columns_to_integers:
    df_movie_budgets[column] = df_movie_budgets[column].astype(str).str.replace(",", '').str.replace("$", "").astype(np.int)

df_movie_budgets

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

      production_budget  domestic_gross  worldwide_gross  
0             425000000       760507625       2776345279  
1             410600000       241063875       1045663875  
2             350000000        42762350        149762350  
3             330600000       459005868       1403013963  
4             317000000       620181382       1316721747  
...                 ...             ...              ...  
5777               7000               0                0  
5778               6000           48482           240495  
5779               5000            1338             1338  
5780               1400               0                0  
5781               1100          181041           181041  

[5782 rows x 6 columns]

In [64]:
# Create year column

df_movie_budgets["year"] = df_movie_budgets["release_date"].str[-4:].astype(int)

df_movie_budgets["year"]

0       2009
1       2011
2       2019
3       2015
4       2017
        ... 
5777    2018
5778    1999
5779    2005
5780    2015
5781    2005
Name: year, Length: 5782, dtype: int64

In [65]:
df_movie_budgets[df_movie_budgets["movie"].str.contains("Avengers")]

id  release_date                    movie  production_budget  \
3     4   May 1, 2015  Avengers: Age of Ultron          330600000   
6     7  Apr 27, 2018   Avengers: Infinity War          300000000   
26   27   May 4, 2012             The Avengers          225000000   
934  35  Aug 14, 1998             The Avengers           60000000   

     domestic_gross  worldwide_gross  year  
3         459005868       1403013963  2015  
6         678815482       2048134200  2018  
26        623279547       1517935897  2012  
934        23385416         48585416  1998

In [66]:
# is id different from movie_id ? 

df_tmdb_movies[df_tmdb_movies["id"] == 27]

# id from df_movie_budgets and df_tmdb_movies don't match. Title is 9 songs when expected The Avengers

genre_ids  id original_language original_title  popularity  \
117  [18,  10402,  10749]  27                en        9 Songs      10.332   

    release_date    title  vote_average  vote_count  
117   2004-09-09  9 Songs           4.9         170

## Data Preparation

Describe and justify the process for preparing the data for analysis.

***
Questions to consider:
* Were there variables you dropped or created?
* How did you address missing values or outliers?
* Why are these choices appropriate given the data and the business problem?
***

Using a Google sheet, I have listed all column names from all dataframes/tables to identify the ones I wanted to keep for the analysis.

The table df_movie_budgets seem to have all the necessary information to calculate performance: 

    1. domestic
    2. foreign 
    3. worldwide gross
    4. production budgets

It was verified before that how titles are named in df_movie_gross don't represent the majority of how they are named in other tables, whereas df_movie_budgets do.

Let's verify that worldwide gross indicated in df_movie_budgets correspond to the one indicated in df_movie_gross 

In [67]:
# Which table to use to merge movie_id: movie_basics or movie_akas? 
    
# Trying to filter on Avengers movie to see if titles are named the same. 
# Avengers in df_movie_gross
df_movie_gross[df_movie_gross["title"].str.contains("Avengers")]

title studio  domestic_gross foreign_gross  year
727     Marvel's The Avengers     BV     623400000.0     895500000  2012
1875  Avengers: Age of Ultron     BV     459000000.0     946400000  2015
3079   Avengers: Infinity War     BV     678800000.0       1,369.5  2018

In [68]:
# Avengers in movie_basics
movie_basics[movie_basics["primary_title"].str.contains("avengers")]

# Not found

movie_id      primary_title          original_title  start_year  \
9562   tt1564369         Scavengers              Scavengers        2013   
20812  tt1864488  Scavengers' Union  Somateio rakosyllekton        2011   
48005  tt2849634     Los Scavengers          Los Scavengers        2014   

       runtime_minutes            genres  
9562              94.0            Sci-Fi  
20812             74.0  Documentary,News  
48005             82.0  Adventure,Family

In [69]:
movie_non_null = movie_basics[movie_basics["original_title"].notnull()]

movie_non_null[movie_non_null["original_title"].str.contains("Avengers")]

movie_id                    primary_title  \
356      tt0848228                     The Avengers   
1030    tt10075836   The Avengers Assemble Premiere   
39010    tt2395427          Avengers: Age of Ultron   
72820    tt4154756           Avengers: Infinity War   
72821    tt4154796                Avengers: Endgame   
105584   tt6172666  Avengers of Justice: Farce Wars   
138087   tt8875872            Avengers: Reassembled   
141910   tt9303032                 Avengers: Legacy   

                         original_title  start_year  runtime_minutes  \
356                        The Avengers        2012            143.0   
1030     The Avengers Assemble Premiere        2012              NaN   
39010           Avengers: Age of Ultron        2015            141.0   
72820            Avengers: Infinity War        2018            149.0   
72821                 Avengers: Endgame        2019            181.0   
105584  Avengers of Justice: Farce Wars        2018             87.0   
138087            Avengers: Reassembled        2018             52.0   
141910                 Avengers: Legacy        2019              NaN   

                         genres  
356     Action,Adventure,Sci-Fi  
1030                     Action  
39010   Action,Adventure,Sci-Fi  
72820   Action,Adventure,Sci-Fi  
72821   Action,Adventure,Sci-Fi  
105584     Action,Comedy,Family  
138087                   Sci-Fi  
141910                   Sci-Fi

In [70]:
# Avengers in movie_akas

movie_akas_df[(movie_akas_df["title"].str.contains("Avengers")) & (movie_akas_df["is_original_title"] == 1)]



movie_id  ordering                    title region language  \
69881   tt2395427        24  Avengers: Age of Ultron   None     None   
81106   tt0848228        19             The Avengers   None     None   
190575  tt4154796        15        Avengers: Endgame   None     None   
311338  tt4154756        30   Avengers: Infinity War   None     None   

           types attributes  is_original_title  
69881   original       None                1.0  
81106   original       None                1.0  
190575  original       None                1.0  
311338  original       None                1.0

In [71]:
# movie title in movie_budgets

df_movie_budgets[df_movie_budgets["movie"].str.contains("Avengers")]

id  release_date                    movie  production_budget  \
3     4   May 1, 2015  Avengers: Age of Ultron          330600000   
6     7  Apr 27, 2018   Avengers: Infinity War          300000000   
26   27   May 4, 2012             The Avengers          225000000   
934  35  Aug 14, 1998             The Avengers           60000000   

     domestic_gross  worldwide_gross  year  
3         459005868       1403013963  2015  
6         678815482       2048134200  2018  
26        623279547       1517935897  2012  
934        23385416         48585416  1998

In [72]:
df_movie_budgets[df_movie_budgets["movie"].str.contains("The Avengers")]

id  release_date         movie  production_budget  domestic_gross  \
26   27   May 4, 2012  The Avengers          225000000       623279547   
934  35  Aug 14, 1998  The Avengers           60000000        23385416   

     worldwide_gross  year  
26        1517935897  2012  
934         48585416  1998

In [124]:
df_movie_gross[df_movie_gross["title"].str.contains("The Avengers")]

title studio  domestic_gross  foreign_gross  year  \
727  Marvel's The Avengers     BV     623400000.0    895500000.0  2012   

      total_gross  total_gross (in '000)  
727  1.518900e+09              1518900.0

In [78]:
movie_basics.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [127]:
# First, try to add movie_id to df_movie_gross
new_df = pd.merge(df_movie_budgets, movie_basics, how="left", left_on=["movie", "year"], right_on=["primary_title", "start_year"])
new_df

# using left join because can only measure the movies that have revenue associated to it



id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5840  78  Dec 31, 2018                                       Red 11   
5841  79   Apr 2, 1999                                    Following   
5842  80  Jul 13, 2005                Return to the Land of Wonders   
5843  81  Sep 29, 2015                         A Plague So Pleasant   
5844  82   Aug 5, 2005                            My Date With Drew   

      production_budget  domestic_gross  worldwide_gross  year   movie_id  \
0             425000000       760507625       2776345279  2009        NaN   
1             410600000       241063875       1045663875  2011  tt1298650   
2             350000000        42762350        149762350  2019  tt6565702   
3             330600000       459005868       1403013963  2015  tt2395427   
4             317000000       620181382       1316721747  2017        NaN   
...                 ...             ...              ...   ...        ...   
5840               7000               0                0  2018        NaN   
5841               6000           48482           240495  1999        NaN   
5842               5000            1338             1338  2005        NaN   
5843               1400               0                0  2015        NaN   
5844               1100          181041           181041  2005        NaN   

                                    primary_title  \
0                                             NaN   
1     Pirates of the Caribbean: On Stranger Tides   
2                                    Dark Phoenix   
3                         Avengers: Age of Ultron   
4                                             NaN   
...                                           ...   
5840                                          NaN   
5841                                          NaN   
5842                                          NaN   
5843                                          NaN   
5844                                          NaN   

                                   original_title  start_year  \
0                                             NaN         NaN   
1     Pirates of the Caribbean: On Stranger Tides      2011.0   
2                                    Dark Phoenix      2019.0   
3                         Avengers: Age of Ultron      2015.0   
4                                             NaN         NaN   
...                                           ...         ...   
5840                                          NaN         NaN   
5841                                          NaN         NaN   
5842                                          NaN         NaN   
5843                                          NaN         NaN   
5844                                          NaN         NaN   

      runtime_minutes                    genres  
0                 NaN                       NaN  
1               136.0  Action,Adventure,Fantasy  
2               113.0   Action,Adventure,Sci-Fi  
3               141.0   Action,Adventure,Sci-Fi  
4                 NaN                       NaN  
...               ...                       ...  
5840              NaN                       NaN  
5841              NaN                       NaN  
5842              NaN                       NaN  
5843              NaN                       NaN  
5844              NaN                       NaN  

[5845 rows x 13 columns]

In [128]:
# verify that avatar does not have a start date on movie_basics
movie_basics[movie_basics["primary_title"].notnull().astype(str).str.contains("(?i)avatar")]

Empty DataFrame
Columns: [movie_id, primary_title, original_title, start_year, runtime_minutes, genres]
Index: []

In [129]:
df_movie_gross[df_movie_gross["title"].str.contains("(?i)avatar")]

title studio  domestic_gross foreign_gross  year
128  Avatar: Special Edition    Fox      10700000.0      22500000  2010

In [132]:
movie_akas_df[(movie_akas_df["title"].str.contains("(?i)avatar")) 
              & (movie_akas_df["is_original_title"] == 1)]


movie_id  ordering                   title region language     types  \
48402   tt2271191         1  Avatars as Prostitutes   None     None  original   
60319   tt1757678         4                Avatar 3   None     None  original   
62979   tt1630029         7                Avatar 2   None     None  original   
109149  tt1854506         2      Aliens vs. Avatars   None     None  original   
151589  tt0253093         1            Gangavataran   None     None  original   
200643  tt3095356         3                Avatar 4   None     None  original   

       attributes  is_original_title  
48402        None                1.0  
60319        None                1.0  
62979        None                1.0  
109149       None                1.0  
151589       None                1.0  
200643       None                1.0

In [30]:
# Here you run your code to clean the data


## Data Modeling
Describe and justify the process for analyzing or modeling the data.

***
Questions to consider:
* How did you analyze or model the data?
* How did you iterate on your initial approach to make it better?
* Why are these choices appropriate given the data and the business problem?
***

In [31]:
# Drop the column Unnamed: 0 with axis=1
# Example heroes_df = heroes_df.drop(['Unnamed: 0'], axis=1)
# heroes_df.head()

# tn.movie_budgets: turn object columns as numbers and strip $ and , 

In [32]:
# Here you run your code to model the data


## Evaluation
Evaluate how well your work solves the stated business problem.

***
Questions to consider:
* How do you interpret the results?
* How well does your model fit your data? How much better is this than your baseline model?
* How confident are you that your results would generalize beyond the data you have?
* How confident are you that this model would benefit the business if put into use?
***

## Conclusions
Provide your conclusions about the work you've done, including any limitations or next steps.

***
Questions to consider:
* What would you recommend the business do as a result of this work?
* What are some reasons why your analysis might not fully solve the business problem?
* What else could you do in the future to improve this project?
***

In [33]:
# Closing connection
# conn.close()